pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -788,3 +857,120 @@ func (p *staticPolicy) GetAllocatableMemory(s state.State) []state.Block {
 	}
 	return allocatableMemory
 }
+
+func (p *staticPolicy) updateStateToReuse(s state.State, pod *v1.Pod, container *v1.Container, memoryBlocks []state.Block) {
```

Comment:
If I understand things correctly, this function is actually doing 2 things.
1. If the current container is an `init` container, assume the "original" state file (i.e. before any memory allocations have been made).
2. If it is an app container, update `reusableState` to reflect the new memory blocks allocated to it.

I guess the assumption here is that we are only maintaining this "temporary" state file so that init containers can pretend that no memory has been allocated to any other containers since they always run 1-by-1 to completion (so have all memory available to them when they are about to run). Once we start considering `app` containers, we have to track it differently because the memory they get allocated cannot be reused.

Once we are working with `app` containers, is there any point in time that `reusableState` != `machineState`? It feels as if things are being overcomplicated here to track this state, if it's really just about making sure that `init` containers see the original memory state. Maybe there's something better we can do to make it clearer.